gateway/ln-gateway/src/lib.rs

```diff
@@ -406,9 +400,16 @@ impl Gateway {
                                             lightning_alias,
                                         }).await;
 
-                                        // Blocks until the connection to the lightning node breaks
-                                        self.handle_htlc_stream(stream, handle.clone()).await;
-                                        warn!("HTLC Stream Lightning connection broken. Gateway is disconnected");
+                                        // Blocks until the connection to the lightning node breaks or we receive the shutdown signal
+                                        tokio::select! {
+                                            _ = self.handle_htlc_stream(stream, handle.clone()) => {
+                                                warn!("HTLC Stream Lightning connection broken. Gateway is disconnected");
+                                            },
+                                            _ = handle.make_shutdown_rx().await => {
+                                                info!("Received shutdown signal");
+                                                break;
```

Comment:
Ahh nice, thanks for adding a check for a shutdown signal on the HTLC stream 👍 

If we break the loop here, we won't hit the codepath to set the gateway state to disconnected and to shutdown the htlc task group.

https://github.com/douglaz/fedimint/blob/fix_tests/gateway/ln-gateway/src/lib.rs#L424-L427

Perhaps we should create a `handle_disconnect` function that's called before the break and also at the end of the loop?

Reply:
+1. I'd be in favor of creating more helper functions here, maybe everything inside of `Ok((stream, ln_client))` can be a function. The diff is just getting hard to read

Reply:
Added a `handle_disconnect`